pkg/resource/predicates.go

```diff
@@ -40,8 +40,8 @@ func NewPredicates(fn PredicateFn) predicate.Funcs {
 	}
 }
 
-// HasClassReferenceKind accepts ResourceClaims that reference the correct kind of resourceclass
-func HasClassReferenceKind(m manager.Manager, k ClassKinds) PredicateFn {
+// HasClassReferenceKinds accepts ResourceClaims that reference the correct kind of resourceclass
+func HasClassReferenceKinds(c client.Client, scheme runtime.ObjectCreater, k ClassKinds) PredicateFn {
```

Comment:
What do you think of naming the `runtime.ObjectCreater` parameter `oc`? It's not necessarily a scheme, and I think we've used `oc runtime.ObjectCreater` as a parameter elsewhere in the codebase.